libp2p/Common.h

```diff
@@ -245,10 +245,9 @@ class Node
     Node(Public _publicKey, NodeIPEndpoint const& _ip, PeerType _peerType = PeerType::Optional): id(_publicKey), endpoint(_ip), peerType(_peerType) {}
     Node(NodeSpec const& _s, PeerType _peerType = PeerType::Optional);
 
-    virtual NodeID const& address() const { return id; }
-    virtual Public const& publicKey() const { return id; }
-    
-    virtual operator bool() const { return (bool)id; }
+    NodeID const& address() const { return id; }
+
+    explicit operator bool() const { return bool{id}; }
```

Comment:
Did the pattern `if (auto n = getNode())` not require change after this?

Reply:
No because `explicit operator bool` can be used in `if` context.